### ISSUE TITLE
4 lock types can be easily removed via UI

### ISSUE THREAD

**Issue by iamsterdam800** [polotno-project/polotno-board#65]
I need to lock some objects in the starter template that I provide to the user.
But users can easily disable those locks.

![image](https://user-images.githubusercontent.com/34508588/232789293-52c97e31-1307-4d3e-a41d-e456303fc313.png)

- element with `removable` lock can be still deleted via Layers panel (the delete button is enabled there)
- to remove `draggable`, `resizable` and `contentEditable` locks all you need to do is to lock the element with the lock button (either in Layers, or in the topbar) and unlock it back. all the locks will be removed from this element!

Only `selectable` lock works.

![image](https://user-images.githubusercontent.com/34508588/232791133-efb5cc6f-c77c-4ccb-9106-4ce3576745a2.png)

Reference polotno export with different locks:
[polotno_locks.json.zip](https://github.com/polotno-project/polotno-board/files/11262066/polotno_locks.json.zip)

**Comment by lavrton** [comment]
> element with removable lock can be still deleted via Layers panel (the delete button is enabled there)

This is a bug. Thanks for pointing into it.

> to remove draggable, resizable and contentEditable locks all you need to do is to lock the element with the lock button (either in Layers, or in the topbar) and unlock it back. all the locks will be removed from this element!

For now, this is intentional. Like, a user can lock/unlock elements when they need that.
Only "selectable: false" is a total lock when a user can't even select an element to do anything with it.

Do you have a use case when you want to lock something buy still allow selection? I would like to understand better.

**Comment by iamsterdam800** [comment]
Yes, in the usecase with the PageView elements, that I explained [here](https://github.com/polotno-project/polotno-board/issues/64#issuecomment-1509901394), I need to lock the PageView elements from moving or resizing or changing or deleting, because they are unchandable part of the final disign. At the same time I want the users to select them, so that for example I can show stuff on the Topbar.

> For now, this is intentional. Like, a user can lock/unlock elements when they need that.

To be honest, I don't see why general lock/unlock should remove all the custom locks. It should keep the previously set custom locks as they were.
The user can not set custom locks themself. So if I set the custom locks in, I did it for a reason.

Maybe if you want to give a user possibility to override custom locking, provide a flag for it?
But I really don't see it as default behaviour.

**Comment by lavrton** [comment]
I would need to deeply think about it. Your use case does make sense to me. Currently, there are no "custom locks". Locks that you change from API and locks that can toggle the user are the same locks. Probably I can add "lockChangeEnabled", that will prevent changing locking from UI.

**Comment by iamsterdam800** [comment]
Yeah, that's one way..
Or (maybe cleaner) just add field `generalLock`.
When you change locking from UI, you only toggle the `generalLock`
And when you check the other locks in the code, you do e.g.:
`if (draggable && !generalLock) { /* ok to change */ }`
is that an idea?
